web/app/js/components/GrafanaLink.jsx

```diff
@@ -3,30 +3,56 @@ import React from 'react';
 import _isEmpty from 'lodash/isEmpty';
 import { grafanaIcon } from './util/SvgWrappers.jsx';
 
-const GrafanaLink = function({ PrefixedLink, name, namespace, resource }) {
+const GrafanaLink = function({ PrefixedLink, name, namespace, resource, grafanaExternalUrl, grafanaPrefix }) {
   let link = `/grafana/d/linkerd-${resource}?var-${resource}=${name}`;
+
+  if (grafanaExternalUrl !== '') {
+    // grafanaPrefix is used for externally hosted Grafana instances, which do not use the /grafana proxy.
+    // When dashboards for multiple Linkerd instances are deployed to the same Grafana host,
+    // the dashboard UID needs a user-specified prefix to be unique.
+    link = `${grafanaExternalUrl}/d/${grafanaPrefix}-linkerd-${resource}?var-${resource}=${name}`;
```

Comment:
If `grafanaPrefix` is empty the uid will get prefixed with `-`. I think if people want to separate the prefix with a dash, they should include it in `grafanaPrefix`.
```suggestion
    link = `${grafanaExternalUrl}/d/${grafanaPrefix}linkerd-${resource}?var-${resource}=${name}`;
```

Reply:
Good idea, I have removed the `-`